### ISSUE TITLE
[Bug]: Excluded routes still appear in the no-config sidebars

### ISSUE THREAD

**Issue by Tyneor** [web-infra-dev/rspress#848]
### Version

```sh
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i5-10310U CPU @ 1.70GHz
    Memory: 2.99 GB / 15.69 GB
  Browsers:
    Chrome: 121.0.6167.185
    Edge: Chromium (122.0.2365.92)
    Internet Explorer: 11.0.19041.3636
  npmPackages:
    rspress: ^1.0.0 => 1.15.1
```

### Details

I added the following lines to the rspress.config.ts :
```
route: {
    exclude: ["assets/**/*", "**/4-ignore.*"],
  },
```

expecting all assets in /docs and the 4-ignore.md not to have routes created for them and not appear in sidebars.

The routes are indeed not created but they still appear in the sidebar:

![image](https://github.com/web-infra-dev/rspress/assets/35744626/c9ddb029-adeb-4fe5-938f-0b47e0cd4f68)

I don't really know if this is a bug but I would say that no-config sidebars should not display non-existing pages.

### Reproduce link

https://codesandbox.io/p/devbox/minimal-route-exclude-in-sidebar-reproduction-gwzr65?file=%2Frspress.config.ts

### Reproduce Steps

1. remove `_meta.json`in docs/guide
2. add assets in docs/guide/assets/
3. add some markdown files in /docs/guide including `4-ignore.md`
4. in rspress.config.ts add 
```
route: {
    exclude: ["assets/**/*", "**/4-ignore.*"],
  },
```
5. the assets and the Ignore route still appear in the guide sidebar

**Comment by 10Derozan** [comment]
Yes, you are right. When fallback to readdir, we need add exclude. Welcome PR

**Comment by 10Derozan** [comment]
Reply to your side note:
`config` hooks can not determine the order, maybe you need to add `sidebar` to avoid internal plugin.
